horace_core/sqw/file_io/@faccess_sqw_v3/private/put_instr_sampl_info_.m

```diff
@@ -73,16 +73,9 @@
         end
     end
 end
-if isa(the_obj, 'unique_objects_container')
-    if the_obj.n_unique == 1 &&  the_obj.n_runs == 1
-        contents = the_obj.unique_objects;
-    elseif the_obj.n_unique > 1 && (the_obj.n_unique == the_obj.n_runs)
-        contents = the_obj.unique_objects;
-        contents = contents(the_obj.idx);
-    else
-        ic = 1:the_obj.n_runs;
-        contents = arrayfun(@(x)the_obj(x),ic,'UniformOutput',false);
-    end
+if isa(the_obj, 'unique_references_container')
+    ic = 1:the_obj.n_runs;
+    contents = arrayfun(@(x)the_obj(x),ic,'UniformOutput',false);
```

Comment:
the "get_all" here has the specific aim of making a cell array of the objects with all the unique objects expanded to duplicates. The set_all dealt with setting the container with a single value. They are not exactly mirror images and they (or at least their functional surrounds) are not my code. It would be easier to leave this for a considered refactor much later.